### ISSUE TITLE
Screen reader issue when using Menu with Virtualizer

### ISSUE THREAD

**Issue by joonass-visma** [adobe/react-spectrum#9452]
### Provide a general summary of the issue here

When trying to use Menu with Virtualizer, screen reader (NVDA) reads the position of currently focused menu item incorrectly.

### 🤔 Expected Behavior?

When going through virtualized list of menu items with keyboard arrows keys, screen reader should announce the position of the currently focused item correctly, for example "Item 1 1 of 600", " Item 2 2 of 600", "Item 3 3 of 600" etc.

### 😯 Current Behavior

It announces the items correctly at first: "Item 1 1 of 600", " Item 2 2 of 600", "Item 3 3 of 600" etc., but when you surpass the point when list scroll happens it gets messed up and says "Item 8 7 of 600".

### 💁 Possible Solution

Should probably +1 the index here because 0 is invalid value for `aria-posinset`:

```javascript
// useMenuItem.ts

if (isVirtualized) {
    ariaProps['aria-posinset'] = item?.index;
    ariaProps['aria-setsize'] = getItemCount(state.collection);
}
```

### 🖥️ Steps to Reproduce

```javascript
<MenuTrigger>
    <Button>Trigger</Button>
    <Popover>
      <Virtualizer layout={ListLayout} layoutOptions={{estimatedRowHeight: 36}}>
          <Menu>
              <MenuItem isVirtualized>Item 1</MenuItem>
              <MenuItem isVirtualized>Item 2</MenuItem>
              {/* + other 598 MenuItems */}
          </Menu>
      </Virtualizer>
    </Popover>
</MenuTrigger>
```

### Version

1.13.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows 11

**Comment by AryanBagade** [comment]
hey @devongovett, would love to work on it, been contributing to react-spectrum. Could you please assign me this issue !

**Comment by snowystinger** [comment]
Look forward to the PR ❤️ ! No need to assign.
